src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -371,7 +372,9 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
     Index getIndex() const;
     void setIndex(const Columns & cols_);
     void setIndex(Columns && cols_);
+    void loadIndex() const TSA_REQUIRES(index_mutex);
     void unloadIndex();
+    void loadIndexWithLock() const { std::scoped_lock lock(index_mutex); loadIndex(); }
```

Comment:
I personally think the `loadIndexWithLock` could be right below `loadIndex` since intuitively they seem to belong together, but it's not that important